doc/io.rst

```diff
@@ -105,6 +105,12 @@ Dataset and DataArray objects, and no array values are loaded into memory until
 you try to perform some sort of actual computation. For an example of how these
 lazy arrays work, see the OPeNDAP section below.
 
+There may be minor differences in the :py:class:`Dataset` object returned
+when reading a NetCDF file with different engines.  For example,
+single-valued attributes are returned as scalars by the default
+``engine=netcdf4``, but as arrays of size ``(1,)`` when reading with
+``engine=h5netcdf``.
```

Comment:
I'm not 100% sure we need to mention this, I think it sort of goes without saying that different backends may differ in minor ways.

Reply:
As a user who understands much less deeply how backends interact with xarray, it did surprise me.

Reply:
It makes total sense but I wasn't aware either, so I'd leave it.